package/kedro_viz/api/rest/router.py

```diff
@@ -49,6 +50,18 @@ async def get_single_node_metadata(node_id: str):
     return get_node_metadata_response(node_id)
 
 
+@router.post("/preferences")
+async def update_preferences(preferences: UserPreference):
+    try:
+        DataNodeMetadata.set_is_all_previews_enabled(preferences.showDatasetPreviews)
+        return {"message": "Preferences updated successfully"}
+    except Exception as exception:
+        logger.error("Failed to update preferences: %s", str(exception))
+        raise HTTPException(
```

Comment:
since we are handling the exception, it would be nice to return a JSONResponse as below -
return JSONResponse(status_code=500, content={"message": f"{exc}"}) if we want the users to see the err trace message on the console. If not we can have a generic message and just log the exception.